source/common/router/header_formatter.cc

```diff
@@ -28,23 +29,27 @@ std::string formatUpstreamMetadataParseException(const std::string& params,
   return fmt::format("Incorrect header configuration. Expected format "
                      "UPSTREAM_METADATA([\"namespace\", \"k\", ...]), actual format "
                      "UPSTREAM_METADATA{}{}",
-                     params, reason);
+                     absl::StrCat(params), reason);
 }
 
 // Parses the parameters for UPSTREAM_METADATA and returns a function suitable for accessing the
 // specified metadata from an RequestInfo::RequestInfo. Expects a string formatted as:
 //   (["a", "b", "c"])
 // There must be at least 2 array elements (a metadata namespace and at least 1 key).
 std::function<std::string(const Envoy::RequestInfo::RequestInfo&)>
-parseUpstreamMetadataField(const std::string& params_str) {
+parseUpstreamMetadataField(absl::string_view params_str) {
+  params_str = StringUtil::trim(params_str);
   if (params_str.empty() || params_str.front() != '(' || params_str.back() != ')') {
     throw EnvoyException(formatUpstreamMetadataParseException(params_str));
   }
 
+  absl::string_view json = params_str;
```

Comment:
You could also write this
 absl::string_view json = params_str.substr(1, params_str.size() - 2);  // trim parens.